File: booking-service/src/main/java/com/example/bookingservice/entity/Ticket.java
package com.example.bookingservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "tblticket")
@Getter
@Setter
@NoArgsConstructor
public class Ticket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tblUserid", nullable = false)
    private Long userId;

    @Column(name = "tblCustomerid", nullable = false)
    private Long customerId;

    @Column(name = "datePayment", nullable = false)
    private LocalDate datePayment;

    @Column(nullable = false, length = 50)
    private String status;

    @Column(length = 500)
    private String note;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal total;

    @OneToMany(mappedBy = "ticket", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ScheduleTicket> scheduleTickets = new ArrayList<>();
}

File: client/src/model/Ticket.js
import { ScheduleTicket } from './ScheduleTicket';

export class Ticket {
    constructor(id = null, userId = null, customerId = null, datePayment = null, status = 'PENDING', note = '', total = 0, scheduleTickets = []) {
        this.id = id;
        this.userId = userId;
        this.customerId = customerId;
        this.datePayment = datePayment;
        this.status = status;
        this.note = note;
        this.total = total;
        this.scheduleTickets = scheduleTickets;
    }
}
